components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java

```diff
@@ -86,6 +99,43 @@ public void handleEvent(Event event) throws IdentityEventException {
                 log.error(errorMsg, e);
                 throw new IdentityEventException(errorMsg);
             }
+
+        } else if (IdentityEventConstants.Event.POST_UPDATE_USER_LIST_OF_ROLE.equals(event.getEventName())) {
+
+            Object userIdList = event.getEventProperties()
+                    .get(IdentityEventConstants.EventProperty.DELETE_USER_ID_LIST);
+            List<String> deletedUserIDList;
+
+            if (userIdList != null) {
+                if (userIdList instanceof List<?>) {
+                    deletedUserIDList = (List<String>) userIdList;
+                    terminateSession(deletedUserIDList);
+                } else {
+                    String errorMsg = "Error occurred due to invalid format of deleted user Id list.";
+                    throw new IdentityEventException(errorMsg);
+                }
+            }
+
+        } else if (IdentityEventConstants.Event.PRE_DELETE_ROLE.equals(event.getEventName()) ||
```

Comment:
let's revoke session on POST_DELETE_ROLE operation

Reply:
Have to go with the `PRE_DELETE_ROLE` as after role deletion, cannot retrieve the user Id list to revoke tokens and terminate sessions.